kernel/kernel-benchmarks/src/test/java/io/delta/kernel/benchmarks/workloadrunners/WriteRunner.java

```diff
@@ -67,14 +67,13 @@ public WriteRunner(WriteSpec workloadSpec, Engine engine) {
   public void setup() throws Exception {
     String tableRoot = workloadSpec.getTableInfo().getResolvedTableRoot();
 
-    // Get the current snapshot
-    SnapshotBuilder builder = TableManager.loadSnapshot(tableRoot);
-    currentSnapshot = builder.build(engine);
-
     // Capture initial listing of delta log files. This is used during cleanup to revert changes.
-    initialDeltaLogFiles = captureFileListing();
+    if (!initialDeltaLogFiles.isPresent()) {
+      initialDeltaLogFiles = Optional.of(captureFileListing());
+    }
+
+    currentSnapshot = loadSnapshot(engine, workloadSpec.getTableInfo(), Optional.empty());
```

Comment:
would `targetSnapshot` be a better name? what does "current" mean in this context?

Reply:
For write specs that have multiple commits, we repeatedly update snapshot. I'll rename it to snapshot for now.


Note: We can benchmark workloads that do a single commit at a time by defining a writeSpec that only contains that one commit.